scene/resources/animation.cpp

```diff
@@ -5594,7 +5594,10 @@ Variant Animation::array_to_string(const Variant p_value) {
 }
 
 Variant Animation::add_variant(const Variant &a, const Variant &b) {
-	if (a.get_type() != b.get_type() && !a.is_array()) {
+	if (a.is_num() && b.is_num()) {
```

Comment:
```suggestion
	if ((a.get_type() != b.get_type() && a.is_num() && b.is_num()) {
```
Right? Otherwise why the switch below, this handles `int` and `float` combinations

Reply:
Yes, I wondered how to best do the conditions here. My reasoning was: Since this is a special case, why not just handle it at the start and otherwise fall back to what would have been done otherwise, especially since the int/int case here looked simpler than the more general conversion in the switch below. But you're right that if `a` is an `INT` or a `FLOAT`, then it is never an array and if `b` is of the same type, then it is always a number, so `case Variant::FLOAT` and `case Variant::INT` should be unreachable now (and could be removed to avoid confusion).

I'm happy to go for whatever you consider more readable/consistent. We could also nest both in the `a.get_type() != b.get_type()` since that's part of both branches then.

```c++
	if (a.get_type() != b.get_type()) {
		if (a.is_num() && b.is_num()) {
			const double result = (a.operator double()) + (b.operator double());
			return cast_from_blendwise(result, a.get_type());
		} else if (!a.is_array()) {
			return a;
		}
	}
```